module/Calculator/Pid.cpp

```diff
@@ -1,15 +1,18 @@
 /**
  * @file Pid.cpp
  * @brief PIDを計算するクラス
- * @author Negimarge
+ * @author Negimarge bizyutyu
  */
 
 #include "Pid.h"
 
 PidGain::PidGain(double _kp, double _ki, double _kd) : kp(_kp), ki(_ki), kd(_kd) {}
 
-Pid::Pid(double _kp, double _ki, double _kd, double _targetValue)
-  : gain(_kp, _ki, _kd), preDeviation(0.0), integral(0.0), targetValue(_targetValue)
+Pid::Pid(double _kp, double _ki, double _kd, double _targetValue, bool _isSpeedCalculator)
+  : gain(_kp, _ki, _kd),
+    preDeviation(_isSpeedCalculator ? 0 : _targetValue - double(Measurer::getManageBrightness())),
```

Comment:
```suggestion
Pid::Pid(double _kp, double _ki, double _kd, double _targetValue, double _initialDeviation)
  : gain(_kp, _ki, _kd),
    preDeviation(_initialDeviation),
```
遅いかもしれけど、偏差の初期値渡せば、Mesuarerに値持たせる必要なくて綺麗にできそう

Reply:
回答になっていないかも知れませんが、偏差の初期値を求めるために、前回取得した輝度をどこかが持っておく必要がある気がします（ライントレースの場合）。

Reply:
なるほど。失礼。Mesuarerについては大丈夫です。
ただ、PIDクラスに渡す値としては、SpeedCalculatorから呼び出されているか、より、メンバ(偏差)の初期値を渡す方がいいと思う。